src/library.py

```diff
@@ -910,82 +910,261 @@ def verify(self):
+                        len(df_occ_diff) > 1
+                    ):  # Since first value of .diff() is nan, this means when values don't change
+                        t_unocc_to_occ = df_occ_diff.index.tolist()[1]
                     else:
-                        result_repo.append(
-                            0
-                        )  # fail, zone temperature setpoint was not reset correctly
-                else:  # room is rented out
-                    T_z_hea_occ_set = day.query("O_sch > 0.0")["T_z_hea_set"].max()
-                    T_z_coo_occ_set = day.query("O_sch > 0.0")["T_z_coo_set"].min()
+                        t_unocc_to_occ = day.index[0]
+
+                    t_s_AHU_off_to_on = df_s_AHU_diff.index.tolist()[1]
 
                     if (
-                        day["T_z_hea_set"] < T_z_hea_occ_set - 2.22 + tol_temp
-                    ).all() or (
-                        day["T_z_coo_set"] > T_z_coo_occ_set + 2.22 - tol_temp
-                    ).all():
-                        result_repo.append(
-                            1
-                        )  # pass, confirm the HVAC setpoint control resets when guest room reset when occupants leave the room
+                        pd.Timedelta(t_s_AHU_off_to_on - t_unocc_to_occ).seconds / 3600
+                        > 12
+                    ):
+                        t_length.append(
+                            pd.Timedelta(t_unocc_to_occ - t_s_AHU_off_to_on).seconds
+                        )
                     else:
-                        result_repo.append(
-                            0
-                        )  # fail, reset does not meet the standard or no reset was observed.
-                year_info = day.index.year[0]
+                        t_length.append(
+                            pd.Timedelta(t_s_AHU_off_to_on - t_unocc_to_occ).seconds
+                        )
+
+                    T_oa_dry_repo.append(day.loc[t_s_AHU_off_to_on]["T_oa_dry"])
+                    T_z_measure = day.loc[t_unocc_to_occ]["T_z_measure"]
+                    T_diff_heating.append(
+                        T_z_measure - day.loc[t_unocc_to_occ]["T_z_hea_set"]
+                    )
+                    T_diff_cooling.append(
+                        abs(T_z_measure - day.loc[t_unocc_to_occ]["T_z_coo_set"])
+                    )
+            year_info = day.index.year[0]
+
+        t_length_array = np.array(t_length).reshape(-1, 1)
+
+        if len(self.df["s_AHU"].unique()) == 1 and self.df["s_AHU"].unique() in [0, 1]:
+            if self.df["s_AHU"].unique() == 0:
+                self.df["result"] = 0
+                self.msg = "NO optimum start"
+            else:
+                if (
+                    mutual_info_regression(t_length_array, T_oa_dry_repo) > 0.5
+                    or mutual_info_regression(t_length_array, T_diff_heating) > 0.5
+                    or mutual_info_regression(t_length_array, T_diff_cooling) > 0.5
```

Comment:
mutual_info_regression returns an nd array, not a number. It is probably fine here. Additionally, 0.5 seems large to me. Do you have any reference on deciding the threshold?

Reply:
1. I added [0] to access ```mutual_info_regression``` return value. Thanks for finding this.
2. For the 0.5 thresholds, I tried to find a reference, but I couldn't find any. From your experience, what would be the appropriate threshold value? Should we also determine this value after we apply this algorithm to a few different buildings/datasets?

Reply:
I don't have a good reference for 0.5. Let's see what the test data tell us after you update the calculation logic.

Reply:
Sure. It sounds great.